### PR TITLE
Refactoring round

### PR DESCRIPTION
**Pull request by nikolasrieble**
Curious to get feedback on the refactoring.

On a general note: It would be great to improve variable naming i.e. not using abbreviations, but I am not sure regarding the names/their meaning.

### CODE REVIEW

File: src/power.py [hunk 1]
```diff
@@ -93,15 +84,32 @@ def get_power(stan_model, y1, y2, sample_size, rope_m, rope_sd, max_hdi_width_m,
...
-            logging.info(pd.DataFrame(power, index=['mean', 'CrIlo', 'CrIhi']).T)
+            log_progress(n_sim, power, step_idx)
 
     for k, v in power.items():
         power[k] = [round(e, precision) for e in v]
 
     return power
 
 
+def generate_simulated_data(mcmc_chain, sample_size, step):
```
Comment (volkale) [inline src/power.py]: change to `_generate_simulated_data` to indicate "private" function

File: src/power.py [hunk 2]
```diff
@@ -93,15 +84,32 @@ def get_power(stan_model, y1, y2, sample_size, rope_m, rope_sd, max_hdi_width_m,
...
+    sigma1_val = mcmc_chain['sigma'][step, 0]
+    sigma2_val = mcmc_chain['sigma'][step, 1]
+    nu_val = mcmc_chain['nu'][step]
+    # Generate simulated data:
+    y1_sim = t.rvs(df=nu_val, loc=mu1_val, scale=sigma1_val, size=sample_size)
+    y2_sim = t.rvs(df=nu_val, loc=mu2_val, scale=sigma2_val, size=sample_size)
+    return y1_sim, y2_sim
+
+
+def log_progress(n_sim, power, step_idx):
```
Comment (volkale) [inline src/power.py]: change to `_ log_progress`

File: src/power.py (hunk 1 above)
Comment (nikolasrieble) [inline src/power.py]:
```suggestion
def _generate_simulated_data(mcmc_chain, sample_size, step):
```

File: src/power.py (hunk 2 above)
Comment (nikolasrieble) [inline src/power.py]:
```suggestion
def _log_progress(n_sim, power, step_idx):
```

Comment (nikolasrieble) [reply src/power.py]: Done

File: src/power.py (hunk 1 above)
Comment (nikolasrieble) [reply src/power.py]: Done

**Review by volkale — APPROVED** [review]

**Pull request merged**